akka-docs-dev/rst/scala/code/docs/http/scaladsl/server/WebsocketExampleSpec.scala

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package docs.http.scaladsl.server
+
+import org.scalatest.{ Matchers, WordSpec }
+
+class WebsocketExampleSpec extends WordSpec with Matchers {
+  "core-example" in {
+    pending // compile-time only test
+    import akka.actor.ActorSystem
+    import akka.stream.ActorFlowMaterializer
+    import akka.stream.scaladsl.{ Source, Flow }
+
+    import akka.http.scaladsl.Http
+    import akka.http.scaladsl.model.HttpMethods._
+    import akka.http.scaladsl.model.{ HttpResponse, Uri, HttpRequest }
+    import akka.http.scaladsl.model.ws.UpgradeToWebsocket
+    import akka.http.scaladsl.model.ws.{ TextMessage, Message }
+
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorFlowMaterializer()
+
+    //#websocket-example-using-core
+    // The Greeter WebSocket Service expects a "name" per message and
+    // returns a greeting message for that name
+    val greeterWebsocketService =
+      Flow[Message]
+        .collect {
+          case tm: TextMessage ⇒ TextMessage(Source.single("Hello ") ++ tm.textStream)
```

Comment:
maybe explain a bit how and why this works. I.e. we do not consume the TextMessage fully before starting to send the answer, but we stream it through. This might be a bit confusing for newcomers.